### PR TITLE
Add Istanbul parser

### PR DESCRIPTION
**Pull request by tolgaozgun**

### CODE REVIEW

File: src/main/java/com/afetyardim/afetyardim/service/PeriodicalSpreadSheetUpdater.java [hunk 1]
```diff
@@ -45,4 +46,14 @@ private void parseIzmirSpreadsheet() {
     }
     log.info("Finish izmir spreadsheet parsing");
   }
+
+  private void parseIstanbulSpreadsheet() {
+    log.info("Start Istanbul spreadsheet parsing");
+    try {
+      izmirSitesParser.parseIzmirSpreadsheet();
```
Comment (akyuznurettin) [inline src/main/java/com/afetyardim/afetyardim/service/PeriodicalSpreadSheetUpdater.java]: please use istanbulSitesParser

File: src/main/java/com/afetyardim/afetyardim/service/istanbul/IstanbulSitesParser.java [hunk 2]
```diff
@@ -0,0 +1,344 @@
...
+                }
+
+                String district;
+                try{
+                    district = (String) row.getValues().get(0).get("formattedValue");
+                }catch(Exception ex){
+                    district = null;
+                }
+
+                if (siteName == null || district == null) {
```
Comment (akyuznurettin) [inline src/main/java/com/afetyardim/afetyardim/service/istanbul/IstanbulSitesParser.java]: siteName null check is redundant, please remove it

**Review by akyuznurettin — CHANGES_REQUESTED** [review]

File: src/main/java/com/afetyardim/afetyardim/service/istanbul/IstanbulSitesParser.java [hunk 3]
```diff
@@ -0,0 +1,344 @@
...
+                }
+            }catch (Exception exception){
+                log.warn("Error while parsing row, {}", row, exception);
+            }
+        }
+
+        log.info("Total rows in excel: {}, Total sites in db before: {}, Created site count: {}, Updated site count: {}"
+                ,rows.size(),istanbulSites.size(),newSites.size(),updatedSiteCount);
+        istanbulSites.addAll(newSites);
+        siteService.saveAllSites(istanbulSites);
```
Comment (akyuznurettin) [inline src/main/java/com/afetyardim/afetyardim/service/istanbul/IstanbulSitesParser.java:82]: please ensure that all sites are correct before saving them to database

Comment (tolgaozgun) [reply src/main/java/com/afetyardim/afetyardim/service/istanbul/IstanbulSitesParser.java:82]: Can you elaborate? I think IzmirSiteParser.java also uses this signature

File: src/main/java/com/afetyardim/afetyardim/service/PeriodicalSpreadSheetUpdater.java (hunk 1 above)
Comment (tolgaozgun) [reply src/main/java/com/afetyardim/afetyardim/service/PeriodicalSpreadSheetUpdater.java]: Fixed, had some fork issues, sorry for the wrong reference

File: src/main/java/com/afetyardim/afetyardim/service/istanbul/IstanbulSitesParser.java (hunk 2 above)
Comment (tolgaozgun) [reply src/main/java/com/afetyardim/afetyardim/service/istanbul/IstanbulSitesParser.java]: Fixed

File: src/main/java/com/afetyardim/afetyardim/service/istanbul/IstanbulSitesParser.java (hunk 3 above)
Comment (akyuznurettin) [reply src/main/java/com/afetyardim/afetyardim/service/istanbul/IstanbulSitesParser.java:82]: yeap, it is just about development tests on your local environment while you are debugging. :)

Comment (tolgaozgun) [reply src/main/java/com/afetyardim/afetyardim/service/istanbul/IstanbulSitesParser.java:82]: I had done those before committing but still went ahead with testing create and update. I also tested on a copy of the target spreadsheet to detect edge cases haha no worries 😀

**Review by akyuznurettin — APPROVED** [review]

**Comment by tolgaozgun** [comment]
Bu PR istanbul_test branchine acildigi icin mergeledigimizde test etme imkanimiz var hala, main'e merge etmiyor. Main'e PR'i bu mergelendikten sonra acabilirim
